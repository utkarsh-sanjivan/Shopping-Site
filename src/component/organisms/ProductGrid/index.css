.product-grid {
    width: 70%;
    display: grid;
    grid-template-columns: auto auto auto auto;
    padding: 10px;
    gap: 10px;
}

@media only screen and (max-width: 1400px) {
    .product-grid {
        grid-template-columns: auto auto auto;
    }
}

@media only screen and (max-width: 900px) {
    .product-grid {
        grid-template-columns: auto auto;
    }
}

@media only screen and (max-width: 700px) {
    .product-grid {
        width: 50%;
        grid-template-columns: auto;
    }
}

@media only screen and (max-width: 400px) {
    .product-grid {
        margin: auto;
        width: 80%;
    }
}