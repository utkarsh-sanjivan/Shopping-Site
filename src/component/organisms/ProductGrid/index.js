import React from 'react';
import GridItem from './../../molecules/GridItem/index';
import './index.css';

export default function ProductGrid(props) {
    /*
        Grid Component that holds a particular product.
        It has all the details of product along with price and quantity.
        It also has a 'Add To Cart' button that can be used to add the product to the cart.
    */
    return (
        <div className='product-grid'>
            {props.productList.map(product => 
                <GridItem
                    {...product}
                    key={product.id}
                    handleSelect={props.handleSelect}
                />
            )}
        </div>
    )
}