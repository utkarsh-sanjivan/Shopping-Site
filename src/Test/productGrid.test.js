import React from 'react';
import { create } from "react-test-renderer";
import ProductGrid from '../component/organisms/ProductGrid/index';

let wrapper;
const renderComponent = props => <ProductGrid {...props} />;
const baseProps = {
    productList: [],
    handleSelect: () => {}
};

beforeEach(() => {
  wrapper = create(renderComponent(baseProps));
});

describe('Testing Product Grid Component Rendering', () => {
  it('Product Grid Component should render properly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
