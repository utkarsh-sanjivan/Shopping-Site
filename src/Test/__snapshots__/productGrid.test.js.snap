// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Product Grid Component Rendering Product Grid Component should render properly 1`] = `
<div
  className="product-grid"
/>
`;
